### PR TITLE
Integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by duskotmane**
# Features ⭐ ⭐ ⭐ 

- [ ] Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
 - Before fixing n+1 problems:
    
bef

 - After fixing n+1 problems:
  
after

- [ ] Use Capybara to write integration tests for each view in your project.
    - Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for 
      the things in the page and another test for clicking links on it.

    - User index page:
       - I can see the username of all other users.
       - I can see the profile picture for each user.
       - I can see the number of posts each user has written.
       - When I click on a user, I am redirected to that user's show page.

   - User show page:
     - I can see the user's profile picture.
     - I can see the user's username.
     - I can see the number of posts the user has written.
     - I can see the user's bio.
     - I can see the user's first 3 posts.
     - I can see a button that lets me view all of a user's posts.
     - When I click a user's post, it redirects me to that post's show page.
     - When I click to see all posts, it redirects me to the user's post's index page.

   - User post index page:
     - I can see the user's profile picture.
     - I can see the user's username.
     - I can see the number of posts the user has written.
     - I can see a post's title.
     - I can see some of the post's body.
     - I can see the first comments on a post.
     - I can see how many comments a post has.
     - I can see how many likes a post has.
     - I can see a section for pagination if there are more posts than fit on the view.
     - When I click on a post, it redirects me to that post's show page.

  - Post show page:
     - I can see the post's title.
     - I can see who wrote the post.
     - I can see how many comments it has.
     - I can see how many likes it has.
     - I can see the post body.
     - I can see the username of each commentor.
     - I can see the comment each commentor left.

I am working solo in this project, here is the prove: 
team

### CODE REVIEW

File: spec/views/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    visit user_path(user)
+
+    expect(page).to have_content('4 posts')
+  end
+
+  scenario 'visiting the user show page, you see the 3 most recent post and bio of the user has written..' do
+    visit user_path(user)
+    expect(page).to have_content('He is a good programmar')
+    expect(page).not_to have_content('fist text')
+  end
```
Comment (Ibizugbe) [inline spec/views/user_show_spec.rb:27]: I noticed you didn't implement a test for showing the users first 3 posts, Kindly do well to implement that as it is a requirement😉.
image

File: spec/views/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    expect(page).to have_content('1 post')
+    expect(page).to have_content('Comments: 3')
+    expect(page).to have_content('Likes: 1')
+  end
+
+  scenario "see some of the post's title, body and first comments" do
+    visit user_posts_path(user)
+    expect(page).to have_content('first text')
+    expect(page).to have_content('first comment')
+    expect(page).to have_content("first post's title")
```
Comment (Ibizugbe) [inline spec/views/post_index_spec.rb]: Nice job so far😉. Kindly do well in writing a pagination test, as this is also a requirement👍🏼.
image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @otmaneechchafyky 👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅

- [x] Test cases have been written to check username of all users👍🏼.
- [x] Gem file has capybare gem for testing👍🏼.
- [x] gitflow is used👍🏼


## Required Changes ♻️

- [x] Kindly note that some of your test cases are failing, do well to ensure all tests are passing before proceeding to submit, to avoid blockers😉.
image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/user_show_spec.rb (hunk 1 above)
Comment (duskotmane) [reply spec/views/user_show_spec.rb:27]: I added it 😃

**Review by IngDavidTM — APPROVED** [review]
Hi @otmaneechchafyky 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**